src/cnaas_nms/api/device.py

```diff
@@ -183,12 +195,15 @@ def put(self, device_id):
                 Device.id == device_id).one_or_none()
 
             if not dev:
-                return empty_result(status='error', data=f"No device with id {device_id}")
+                return empty_result(status='error', data=f"No device with id {device_id}"), 404
 
             errors = dev.device_update(**json_data)
             if errors:
-                return empty_result(status='error', data=errors), 404
-            return empty_result(status='success', data={"updated_device": dev.as_dict()}), 200
+                return empty_result(status='error', data=errors), 400
+            session.commit()
+            update_device_primary_groups()
+            dev_dict = device_data_postprocess([dev])[0]
+            return empty_result(status='success', data={"updated_device": dev_dict}), 200
```

Comment:
Perhaps a quirk I don't understand, but empty_result, does not seem that empty ;)

Reply:
Haha yes indeed! A better name could maybe be make_response?